windows-agent/internal/daemon/networking.go

```diff
@@ -0,0 +1,101 @@
+package daemon
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net"
+	"os/exec"
+	"strings"
+
+	log "github.com/canonical/ubuntu-pro-for-wsl/common/grpc/logstreamer"
+)
+
+// getWslIP returns the loopback address if the networking mode is mirrored or iterates over the network adapters to find the IP address of the WSL one.
+func getWslIP(ctx context.Context, i hostIpconfig, w wslSystemDistro) (net.IP, error) {
+	mode, err := networkingMode(ctx, w)
+	if err != nil {
+		log.Warningf(ctx, "could not determine if WSL network is mirrored (assuming NAT): %v", err)
+		mode = "nat"
+	}
+
+	switch mode {
+	case "mirrored":
+		return net.IPv4(127, 0, 0, 1), nil
+	case "nat":
+		return findWslAdapterIP(i)
+	default:
+		return nil, fmt.Errorf("unknown networking mode: %s", mode)
+	}
+}
+
+func findWslAdapterIP(i hostIpconfig) (net.IP, error) {
+	const targetDesc = "Hyper-V Virtual Ethernet Adapter"
+	const vEthernetName = "vEthernet (WSL"
+
+	head, err := i.getAdaptersAddresses()
+	if err != nil {
+		return nil, err
+	}
+
+	// Filter the adapters by description.
+	var candidates []ipAdapterAddresses
+	for node := head; node != nil; node = node.Next() {
+		desc := node.Description()
+		// The adapter description could be "Hyper-V Virtual Ethernet Adapter #No"
+		if !strings.Contains(desc, targetDesc) {
```

Comment:
`targetDesc` and `vEthernetName` are only used once, so having it declared as a const force you to look up while the direct string comparison (even comparing with the method itself) is more straighforward.

What about:
`if !strings.Contains(node.Description(), "Hyper-V Virtual Ethernet Adapter") {` ?

Also, note that I directly used `node.Description()` than assigning a variable that is not going to reused. Btw, you did exactly this with `node.FriendlyName()` few lines below :p